lib/install/management/dispatcher.go

```diff
@@ -125,12 +131,60 @@ func (d *Dispatcher) initDiagnosticLogs(conf *metadata.VirtualContainerHostConfi
 	}
 }
 
+func (d *Dispatcher) registerExtension(conf *metadata.VirtualContainerHostConfigSpec) error {
+	log.Infoln("Registering VCH as a vSphere extension")
+	extensionManager := object.NewExtensionManager(d.session.Vim25())
+	extension := types.Extension{
+		Description: &types.Description{
+			Label:   "VIC",
+			Summary: "vSphere Integrated Containers Virtual Container Host",
+		},
+		Company: "VMware, Inc.",
+		Version: "0.2",
+		Key:     "com.vmware.vic",
+	}
+
+	var err error
+	var certbuffer, keybuffer bytes.Buffer
+	if certbuffer, keybuffer, err = certificate.CreateRawKeyPair(); err != nil {
```

Comment:
@emlin I'm not sure that I understand what you mean. I've moved the certificate routines from cmd/vic-machine/create to their own package so that I can call into them. Right now the generation does take place before the rest of the resource creation, except the resource pool, but I can move the routine to be called before that too. I believe I put it after the resource pool since the creation of the resource pool needs to have a unique name and I plan to reuse that name to make the extension's key unique.